README.md

```diff
@@ -1,289 +1,607 @@
+
+The contents of a secret entry may be anything representable as bytes on the file system including, but not limited to, a literal string value (e.g. `db-password`), a language-specific binary (e.g. a Java `KeyStore` with a private key and X.509 certificate), or an indirect pointer to another system for value resolution (e.g. `vault://production-database/password`).
+
+The collection of files within the directory **MAY** change between container launches.  The collection of files within the directory **SHOULD NOT** change during the lifetime of the container.
+
+## Example Directory Structure
+
+```plain
+$SERVICE_BINDING_ROOT
+├── account-database
+│   ├── kind
+│   ├── connection-count
+│   ├── uri
+│   ├── username
+│   └── password
+└── transaction-event-stream
+│   ├── kind
+│   ├── connection-count
+│   ├── uri
+│   ├── certificates
+│   └── private-key
+```
+
+# Service Binding
+
+A Service Binding describes the connection between a [Provisioned Service](#provisioned-service) and an [Application Projection](#application-projection).  It is codified as a concrete resource type.  Multiple Service Bindings can refer to the same service.  Multiple Service Bindings can refer to the same application.
+
+A Service Binding resource **MUST** define a `.spec.application` which is an `ObjectReference` to a `PodSpec`-able resource.  A Service Binding resource **MUST** define a `.spec.service` which is an `ObjectReference` to a Provisioned Service-able resource.  A Service Binding resource **MAY** define a `.spec.name` which is the name of the service when projected into the application.
+
+A Service Binding resource **MUST** define a `.status.conditions` which is an array of `Condition` objects.  A `Condition` object **MUST** define `type`, `status`, and `lastTransitionTime` entries.  At least one condition containing a `type` of `Ready` must be defined.  The `status` of the `Ready` condition **MUST** have a value of `True`, `False`, or `Unknown`.  The `lastTranstionTime` **MUST** contain the last time that the condition transitioned from one status to another.  A Service Binding resource **MAY** define `reason` and `message` entries to describe the last `status` transition.
+
+## Resource Type Schema
+
+```yaml
 apiVersion: service.binding/v1alpha1
 kind: ServiceBinding
 metadata:
-  name: example-service-binding
+  name:                 # string
 spec:
-  mountPathPrefix: "/var/bindings"
-  services:
-    - group: postgres.dev
-      kind: Service
-      name: global-user-db
-      version: v1beta1
-      id: postgres-global-user
-    - group: ibmcloud.ibm.com
-      version: v1alpha1
-      kind: Binding
-      name: coligo-service-binding
-      id: coligo-service-binding
-  application:
-    name: nodejs-rest-http-crud
-    group: apps
-    version: v1
-    resource: deployments
+  name:                 # string, optional, default: .metadata.name
+  kind:                 # string, optional
+  provider:             # string, optional
```

Comment:
Should we keep the `kind` and `provider` fields in the core spec, or make them extensions? While I think they will be very useful, it adds a good bit of complexity (secret re-projection) to an otherwise conforming implementation that doesn't need them.

Reply:
My reasoning for inclusion in the core spec is the same as `name`; regardless of the implementation, being able to set this here gives an added level of flexibility that is broadly useful.  I'm interested in outsiders' opinions though.

Reply:
I think it's useful to have them in the `ServiceBinding` CR because then it could lower the entry bar for provisioned services to be spec compliant.  If the provisioned service's Secret doesn't specify `kind` or `provider`, that's OK, as the `ServiceBinding` CR can specify them.  

But then we would likely have to relax the restrictions around `kind` a bit - perhaps saying it is required to be present in the persisted / mounted volume, but not required in the provisioned service's Secret.  And if `kind` is not present in either the original Secret nor defined in the `ServiceBinding` CR, we can likely have a `recommended` default of the provisioned service's CR `kind`?

Reply:
That's a great suggestion.  I'll figure out where that goes.